homeassistant/components/template/light.py

```diff
@@ -249,10 +306,33 @@ def template_light_startup(event):
             self._brightness = kwargs[ATTR_BRIGHTNESS]
             optimistic_set = True
 
+        if self._temperature_template is None and ATTR_COLOR_TEMP in kwargs:
+            _LOGGER.info(
+                "Optimistically setting color temperature to %s",
+                kwargs[ATTR_COLOR_TEMP],
+            )
+            self._temperature = kwargs[ATTR_COLOR_TEMP]
+            optimistic_set = True
+
+        if self._color_template is None and ATTR_HS_COLOR in kwargs:
+            _LOGGER.info("Optimistically setting color to %s", kwargs[ATTR_HS_COLOR])
+            self._color = kwargs[ATTR_HS_COLOR]
+            optimistic_set = True
+
         if ATTR_BRIGHTNESS in kwargs and self._level_script:
             await self._level_script.async_run(
                 {"brightness": kwargs[ATTR_BRIGHTNESS]}, context=self._context
             )
+        elif ATTR_COLOR_TEMP in kwargs and self._temperature_script:
+            await self._temperature_script.async_run(
+                {"color_temp": kwargs[ATTR_COLOR_TEMP]}, context=self._context
+            )
+        elif ATTR_HS_COLOR in kwargs and self._color_script:
+            hs_value = kwargs[ATTR_HS_COLOR]
+            await self._color_script.async_run(
```

Comment:
Why are all these different scripts? Since it is all done inside the async_turn_on method, shouldn't there just be a generic turn_on script ?

Reply:
All these scripts map the actions to call when the brightness/color/temperature is updated.